doc/user/content/ingest-data/sizing.md

```diff
@@ -0,0 +1,357 @@
+
+### Workloads
+There are 2 primary workloads that sources currently fall under in Materialize:
+
+- Upsert-like: `ENVELOPE UPSERT` and `ENVELOPE DEBEZIUM` Kafka sources.
+- Append-only: all other sources, including Postgres sources.
+
+These workloads perform significantly differently, and are covered in separate sections below.
+
+
+## Upsert-like sources
+Upsert and Debezium sources are required to store a _local_ version of ALL active key-value pairs the source's working set to work.
+this is because when a key is _updated_, the source must be able to produce a [CDC](https://en.wikipedia.org/wiki/Change_data_capture)-style
+_retraction_ for the previous value. This is a consequence of how data is stored in Materialize and is fundamental to making computations
+on top of sources efficient.
+
+The _local state_ for Upsert and Debezium sources defaults to in-memory, but [Disk-attached replicas](/sql/create-cluster-replica/#disk-attached-replicas)
+move that state onto an attached disk. See (TODO: link to https://github.com/MaterializeInc/materialize/pull/21438 when its merged) for details on how to
+configure that feature for clusters.
+
+### Steady-state
+In the steady-state, upsert-like sources require enough space in their _local state_ to store the entire _working set_. The _working set_
+is the set of all active keys and their current values.
+
+#### Determining your steady-state size
+The first step in creating a source cluster for 1 or more upsert-like sources is to estimate the sum total _working set_ size for all
+sources. This estimate does not have to be perfect.
+
+After determining this estimate, choose the smallest size from [Available sizes](#available-sizes) whose `total_memory_gib` is larger than
+the estimate (use `total_disk_gib` if using disk-attached replicas). This is your `steady_state_size`.
+
+### Hydration and starting your sources
+{{< note >}}
+The limitation discussed in this section may be removed in the future.
+{{</ note >}}
+
+There is currently a limitation in Materialize where _hydration_ uses additional memory (even when using disk-attached replicas) then
+both _steady-state_ and _rehydration_.
+
+#### Determining your hydration size
+The amount of memory required to hydrate a source is _roughly_ equal to the total size of its Kafka topic. You can total this size for
+ALL sources you are going to run on your source cluster, and compare it to `total_memory_gib` in [Available sizes](#available-sizes) to
+determine your `hydration_size`, which may be larger than your `steady_state_size`.
+
+After doing so, create your cluster:
+```sql
+CREATE CLUSTER src_cluster MANAGED, REPLICATION FACTOR = 1, SIZE = <hydration_size>
+```
+
+#### Starting hydration
+You can then run all your `CREATE SOURCE .. IN CLUSTER src_cluster` statements to start the hydration process for all your sources.
+
+#### Waiting for hydration
+There is currently not a good way of determining whether or not hydration is finished. You can run:
```

Comment:
If they know the offset number, can they check the `_progress` table?

Reply:
Yes. This is the better solution.